package-system/python/darwin_x64/open3d_python.patch

```diff
@@ -1,201 +1,22 @@
--                            "-I%s/usr/local/include"%(
--                        recipe.get('extra_cflags', ''),
-+        if not recipe.get('suppressCFlags', False):
-+            if recipe.get('useLDFlags', 1):
-+                configure_args.extend([
-+                    "CFLAGS=%s-mmacosx-version-min=%s -arch %s "
-+                                "-I%s/usr/local/include"%(
-+                            recipe.get('extra_cflags', ''),
-+                            DEPTARGET,
-+                            ' -arch '.join(archList),
-+                            shellQuote(basedir)[1:-1],),
-+                    "LDFLAGS=-mmacosx-version-min=%s -L%s/usr/local/lib -arch %s"%(
-                         DEPTARGET,
--                        ' -arch '.join(archList),
--                        shellQuote(basedir)[1:-1],),
--                "LDFLAGS=-mmacosx-version-min=%s -L%s/usr/local/lib -arch %s"%(
--                    DEPTARGET,
--                    shellQuote(basedir)[1:-1],
--                    ' -arch '.join(archList)),
--            ])
--        else:
--            configure_args.extend([
--                "CFLAGS=%s-mmacosx-version-min=%s -arch %s "
--                            "-I%s/usr/local/include"%(
--                        recipe.get('extra_cflags', ''),
--                        DEPTARGET,
--                        ' -arch '.join(archList),
--                        shellQuote(basedir)[1:-1],),
--            ])
-+                        shellQuote(basedir)[1:-1],
-+                        ' -arch '.join(archList)),
-+                ])
-+            else:
-+                configure_args.extend([
-+                    "CFLAGS=%s-mmacosx-version-min=%s -arch %s "
-+                                "-I%s/usr/local/include"%(
-+                            recipe.get('extra_cflags', ''),
-+                            DEPTARGET,
-+                            ' -arch '.join(archList),
-+                            shellQuote(basedir)[1:-1],),
-+                ])
-
-         if 'configure_post' in recipe:
-             configure_args = configure_args + list(recipe['configure_post'])
-@@ -1090,6 +1092,8 @@ def buildPythonDocs():
-     if not os.path.exists(htmlDir):
-         # Create virtual environment for docs builds with blurb and sphinx
-         runCommand('make venv')
-+        runCommand('venv/bin/python3 -m pip install -U markupsafe==2.0.1')
-+        runCommand('venv/bin/python3 -m pip install -U jinja2==2.11.3')
-         runCommand('venv/bin/python3 -m pip install -U Sphinx==2.3.1')
-         runCommand('make html PYTHON=venv/bin/python')
-     os.rename(htmlDir, docdir)
-@@ -1522,6 +1524,11 @@ def buildInstaller():
-     pkgroot = os.path.join(outdir, 'Python.mpkg', 'Contents')
-     pkgcontents = os.path.join(pkgroot, 'Packages')
-     os.makedirs(pkgcontents)
-+
-+    #o3de modification:  We don't need packages for our purposes, so we return immediately:
-+    return
-+
-+
```

Comment:
so this was all removed ... .. intentionally?  It works without this?  ANd it still correctly sets min OS to the right number?

Reply:
None of the original patch from 3.7 applies to the 3.10 version of [build-installer.py](https://github.com/python/cpython/blob/v3.10.5/Mac/BuildScript/build-installer.py), but I wasn't notice that we were short-circuiting the packages.  Removing that short-circuit didn't seem to have any problems, all the python unit tests passed on Mac for me. I did not check the min OS number

Reply:
The original patch was partially restored (the part that dealt with the mac osx min version), and restored the skipping of the package creation. Verified with otool that it is set on the binary

```
otool -l python3 | grep -E -A4 '(LC_VERSION_MIN_MACOSX|LC_BUILD_VERSION)'
      cmd LC_BUILD_VERSION
  cmdsize 32
 platform 1
    minos 10.15
      sdk 12.1
```

Reply:
I guess 10.15 is okay, but the minos in our toolchain file is actually 11.0
we don't set a SDK version though (But perhaps thats related to minos?)

10.15 would work in all the situations we need it to.  But is it working by accident becuase we happen to be using that SDK to build it?

Reply:
10.15 was specific to the MAc build scripts for python, even on the current version. https://github.com/python/cpython/blob/main/Mac/BuildScript/README.rst